### PR TITLE
Add grants_tagger visualize

### PR DESCRIPTION
**Pull request by nsorros**
Adds `grants_tagger vizualize` command that runs a [streamlit](https://streamlit.io) app. It allows the user to select a model, threshold and whether to display probabilities and then paste any text to get tags

Screenshot 2021-06-28 at 2 06 19 PM

### CODE REVIEW

File: grants_tagger/__main__.py [hunk 1]
```diff
@@ -362,6 +363,10 @@ def explain():
     # feature importance for models
     pass
 
+@app.command()
+def visualize():
+    st_app_path = os.path.join(os.path.dirname(__file__), "streamlit_visualize.py")
+    subprocess.Popen(["streamlit", "run", st_app_path])
```
Comment (aCampello) [inline grants_tagger/__main__.py:369]: Wonder if there is a way to run Streamlit programatically, like Flasks' `import app; app.run()`

File: grants_tagger/streamlit_visualize.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import streamlit as st
+import pandas as pd
+
+from grants_tagger import predict_tags
+
+threshold = st.sidebar.slider("Threshold", min_value=0.0, max_value=1.0, value=0.5)
+text = st.text_area('Grant abstract', 'The cell is...', height=300)
+
+models = {
+    "disease_mesh_cnn-2021.03.1": {
```
Comment (aCampello) [inline grants_tagger/streamlit_visualize.py:10]: Nice model dispatcher, although in the future you can probably accept the model config file as parameter to your `grants_tagger visualize`.

File: grants_tagger/streamlit_visualize.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        "approach": "scibert"
+    }
+}
+
+model_option = st.sidebar.selectbox("Model", options=list(models.keys()))
+model = models[model_option]
+
+probabilities = st.sidebar.checkbox("Display probabilities")
+
+with st.spinner('Calculating tags...'):
```
Comment (aCampello) [inline grants_tagger/streamlit_visualize.py:37]: So easy.

**Review by aCampello — APPROVED** [review]
Code looks good, and from the video/pictures the visualisation looks quite good too! Streamlit syntax seems nice and clean.

Seems to have a conflict with requirements.txt to fix before merging.

File: grants_tagger/__main__.py (hunk 1 above)
Comment (nsorros) [reply grants_tagger/__main__.py:369]: I did not find a way, but will look again. Btw @jdu is this a good way if invoking a script? I remember hunter had mentioned that some python way was better than another

File: grants_tagger/streamlit_visualize.py (hunk 3 above)
Comment (nsorros) [reply grants_tagger/streamlit_visualize.py:37]: <3 streamlit

File: grants_tagger/__main__.py (hunk 1 above)
Comment (jdu) [reply grants_tagger/__main__.py:369]: There's a few different ways of spawning a sub-process across the different python versions. This is probably fine for what is just a labs component.

**Pull request merged**